wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/RdsHostListProvider.java

```diff
@@ -157,31 +147,59 @@ protected void initSettings() throws SQLException {
     this.rdsUrlType = rdsHelper.identifyRdsType(this.initialHostSpec.getHost());
   }
 
+  protected ClusterTopologyMonitor initMonitor() throws SQLException {
+    return this.servicesContainer.getMonitorService().runIfAbsent(
+        ClusterTopologyMonitorImpl.class,
+        this.clusterId,
+        this.servicesContainer,
+        this.properties,
+        (servicesContainer) -> new ClusterTopologyMonitorImpl(
+            servicesContainer,
+            this.topologyUtils,
+            this.clusterId,
+            this.initialHostSpec,
+            this.properties,
+            this.instanceTemplate,
+            this.refreshRateNano,
+            this.highRefreshRateNano));
+  }
+
+  protected List<HostSpec> queryForTopology() throws SQLException {
+    ClusterTopologyMonitor monitor = this.servicesContainer.getMonitorService()
+        .get(ClusterTopologyMonitorImpl.class, this.clusterId);
+    if (monitor == null) {
+      monitor = this.initMonitor();
+    }
+
+    try {
+      return monitor.forceRefresh(false, DEFAULT_TOPOLOGY_QUERY_TIMEOUT_MS);
+    } catch (TimeoutException ex) {
+      return null;
+    }
+  }
+
   /**
    * Get cluster topology. It may require an extra call to database to fetch the latest topology. A
    * cached copy of topology is returned if it's not yet outdated (controlled by {@link
    * #refreshRateNano}).
    *
-   * @param conn        A connection to database to fetch the latest topology, if needed.
    * @param forceUpdate If true, it forces a service to ignore cached copy of topology and to fetch
    *                    a fresh one.
    * @return a list of hosts that describes cluster topology. A writer is always at position 0.
    *     Returns an empty list if isn't available or is invalid (doesn't contain a writer).
    * @throws SQLException if errors occurred while retrieving the topology.
    */
-  protected FetchTopologyResult getTopology(final Connection conn, final boolean forceUpdate) throws SQLException {
-    init();
-
+  protected FetchTopologyResult getTopology(final boolean forceUpdate) throws SQLException {
     final List<HostSpec> storedHosts = this.getStoredTopology();
     if (storedHosts == null || forceUpdate) {
       // We need to re-fetch topology.
-      if (conn == null) {
-        // We cannot fetch the latest topology since we do not have access to a connection, so we return the original
-        // hosts parsed from the connection string.
+      if (!this.pluginService.isDialectConfirmed()) {
```

Comment:
This is needed because without it, cluster topology monitors are created before the db dialect has been updated. This causes errors because the topology monitors are not executing the correct SQL queries.